Core/Controller/EditRegularizacionImpuesto.php

```diff
@@ -206,62 +188,87 @@ protected function execAfterAction($action)
     }
 
     /**
-     * Run the autocomplete action with exercise filter
-     * Returns a JSON string for the searched values.
+     * Load data view procedure
      *
-     * @return array
+     * @param string                      $viewName
+     * @param ExtendedController\BaseView $view
      */
-    protected function autocompleteAction(): array
+    protected function loadData($viewName, $view)
     {
-        $results = [];
-        $data = $this->requestGet(['field', 'source', 'fieldcode', 'fieldtitle', 'term', 'codejercicio']);
-        $fields = $data['fieldcode'] . '|' . $data['fieldtitle'];
-        $where = [
-            new DataBaseWhere('codejercicio', $data['codejercicio']),
-            new DataBaseWhere($fields, mb_strtolower($data['term']), 'LIKE')
-        ];
-
-        foreach (CodeModel::all($data['source'], $data['fieldcode'], $data['fieldtitle'], false, $where) as $row) {
-            $results[] = ['key' => $row->code, 'value' => $row->description];
-        }
-        return $results;
+        $function = 'get' . str_replace("-", "", $viewName);
+        $this->$function($view);
```

Comment:
This solutions works, and has the minimum code. But it's difficult to read and understand. A switch statement is a better solution.

Reply:
Perhaps with a comment that explains it would be enough.